dbms/src/Storages/Page/V3/BlobStore.cpp

```diff
@@ -258,10 +282,26 @@ BlobStore<Trait>::write(typename Trait::WriteBatch & wb, const WriteLimiterPtr &
     if (all_page_data_size == 0)
     {
         // Shortcut for WriteBatch that don't need to persist blob data.
-        for (auto & write : wb.getWrites())
+        for (auto & write : wb.getMutWrites())
         {
             switch (write.type)
             {
+            case WriteBatchWriteType::PUT_REMOTE:
+            {
+                PageEntryV3 entry;
+                entry.file_id = INVALID_BLOBFILE_ID;
+                entry.tag = write.tag;
+                entry.checkpoint_info = CheckpointInfo{
+                    .data_location = *write.data_location,
+                    .is_local_data_reclaimed = true,
+                };
+                if (!write.offsets.empty())
+                {
+                    entry.field_offsets.swap(write.offsets);
```

Comment:
If we use `swap` here, better use `Trait::WriteBatch && wb` because we changed `wb` already.

Reply:
Fixed.